staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -285,10 +285,12 @@ func (s *WatchServer) HandleWS(ws *websocket.Conn) {
 	buf := &bytes.Buffer{}
 	streamBuf := &bytes.Buffer{}
 	ch := s.Watching.ResultChan()
+
+	defer s.Watching.Stop()
```

Comment:
Is it OK to double-stop this thing? (e.g. for line 298 it is likely already stopped)

Reply:
FWIW, that's exactly the path exercised by line 204/226 above in the normal case of the watch channel closing